pkg/controller/applicationmonitoring/applicationmonitoring_controller.go

```diff
@@ -171,6 +174,62 @@ func (r *ReconcileApplicationMonitoring) reconcileConfig(cr *applicationmonitori
 	return reconcile.Result{RequeueAfter: time.Second * ReconcilePauseSeconds}, err
 }
 
+func (r *ReconcileApplicationMonitoring) checkServiceAccountAnnotationsExist(cr *applicationmonitoringv1alpha1.ApplicationMonitoring) (reconcile.Result, error) {
```

Comment:
Can this be generalized to work on any service account instead of handling the alertmanager and prometheus cases explicitely?

Reply:
Think this is completed now, can you take another look @pb82